### ISSUE TITLE
gl proxy mode problem

### ISSUE THREAD

**Issue by emadmahdi** [Shani-08/ShaniXBMCWork#161]
@Shani-08 ... i noticed a problem with glarab proxy mode ... when you enable it using the settings page ... it will always play the channel "KuwaitSpace_Med" and it will always use this ip "38.99.146.71".

Anyways ... I rewrote the whole xml file and proxy mode is now available without setting the proxy ip ... it is now hardcoded in the xml file ... I also included only the LOW quality for the non-proxy method since it is now the only one available ... I also added the LR quality for the proxy mode.

I think it is better to cancel the automatic url proxy change because the list of channels between both proxy and non-proxy methods are not the same that is why I included in the xml file five entries for each channel ... one entry as LOW and four entries as Proxy HD, MED, LOW, & LR.

Currently the proxy ip and the ip of the LR quality is both hardcoded in the xml file ... I would suggest changing them to be copied from the settings page ... i mean similar to the ip's of the LOW, MED & HD.

**Comment by emadmahdi** [comment]
@Shani-08 ... also when adding this ip "38.99.146.71" to the list for MED ... the same problem happen ... it will always play the channel "KuwaitSpace_Med".

**Comment by Shani-08** [comment]
Its playing like that because it tries to read the channel name and it thinks it would be after $ like $/channel.m3u8
but since you have hard coded the ip, its not able to find and fallbacks on that default channel.

Your change has few problems.
1. I dont think we should use 127.0.0.1, there is a setting and it should use because it could be that the proxy is not on local machine, like when you are running it on android or ios.
2. should we be hardcoding the ip address for proxy server for HD links? are these different from when we get the ip from server call when returns the session?

I think we need to split it in two sources , GLArab and GLArab proxy. The first one will use the ip from the config/settings.
GLArab Proxy will have proxy with HD etc links. We can create saparate pool of IP in settings.
Lastly, when we are breaking, i think we should make the xml file without the full link, like
`
<streaminginfo>
	<cname>2M</cname>
	<item>
	<title>Low</title>
	<link>ip=list;path=2M;session=1</link>
	</item>
</streaminginfo>

<streaminginfo>
	<cname>Nile Drama2</cname>
	<item>
	<title>Low</title>
	<link>ip=99.11.44.33;path=NDrama2;session=0</link>
	</item>
</streaminginfo>
`
values for session, 0 means session not required, 1 means session required.
ip will have values
*static ip address
*list= try all ip from the settings
*session= use the ip thats returned from session call
 

For proxy source
`
<streaminginfo>
	<cname>Nile Drama2</cname>
	<item>
	<title>HD</title>
	<link>ip=99.11.44.33;path=NDrama2</link>
	</item>
</streaminginfo>

<streaminginfo>
	<cname>Nile Drama2</cname>
	<item>
	<title>HD</title>
	<link>ip=session;path=NDrama2</link>
	</item>
</streaminginfo>
`

possible values for ip would be, 
*static ip address
*list= try all ip from the settings
*session= use the ip thats returned from session call
 

Perhaps we can start with writing new xml for proxy file, i will do the source and coding and then we just change the current one , remove everyone and leave low channels.

**Comment by Shani-08** [comment]
I have just done the quick fix as it would try using channel= format too. it should be working

**Comment by emadmahdi** [comment]
I think it is better to keep it as simple as possible ... changing the format will add more complexity and will not allow future changes easily ... plus the non-proxy is not really available like before ... it is very limited now (only some LOW channels) ... which changes the whole plugin to be mostly for a proxy method.

For sessionID needed or not ... it depends on the server ... each server has different needs which can be defined with the definition of the ip ... for example the ip's 19 & 20 they do not need any sessionID for all qualities.

For the proxy ip ... yes we do need to change it to a variable ... although the gl proxy accepts requests only from localhost or 127.0.0.1 unless an older version of the proxy is used which unfortunately has problems with audio decoding.

I also suggest adding a settings for the ip of the LR quality.

As for a pool of ip's .... I do recommend to continue using it ... because usually there is at least 2 ip's available for each quality.

As for the three kinds of ip ... I don't think that we need more than the pool ip's ... because the ip that the session will provide is very specific to that channel not for all channels ... as for the static ip I don't know why we will need it since we are the ones who define these ip's in the settings.

Anyways ... if you decided to change the format then let me know and i will rewrite the xml file again ... i will also update the ip list soon.

i also suggest to remove the port number from the ip list for only the proxy method ... and then in the code you can use this change as a flag to distinguish which ip is used for proxy method and which ip is used for non-proxy method.

**Comment by emadmahdi** [comment]
ip list updated except for proxy LR ... it is this ... 38.99.146.152|8.21.48.116|8.21.48.117

**Comment by allouch89** [comment]
@Shani-08 what do you mean by:
1. I dont think we should use 127.0.0.1, there is a setting and it should use because it could be that the proxy is not on local machine, like when you are running it on android or ios.

Is it possible to access the proxy method in another way than using a local machine to host the proxy?

**Comment by allouch89** [comment]
Maybe you guys can help me out, i have installed the gl_proxy on a local machine, and im poiting the 4500 port to it, but i can only access the proxy from the same machine, do i have to use a proxy to connect through for it to work?

**Comment by testiddd** [comment]
another problem is that not all streams for proxy method are called e.g. RotanaClip_HD , if you change it to RotanaClip_High it is working.
There is any other proxy to be used beside these 1.4... versions , that can be used to play content not on the same machine without audio problems ?

**Comment by allouch89** [comment]
where can i get these 1.4 versions?

**Comment by Shani-08** [comment]
@testiddd  you should be able to correct the name in the xml file, if its wrong.

**Comment by Shani-08** [comment]
@emadmahdi i can live with format thats fine. Are we agreeing to split the source in two?
right now, when proxy is enabled, we dont even reading the link as it creates its own so thats why i was thinking why we need it as only thing it needs is channel and ip(if static otherwise it uses session ip)
we can leave session as it is, I dont mind.

**Comment by allouch89** [comment]
Btw id anyone is looking for the old version of the proxy, you can find it here:
https://www.glarab.com:447/zixi/download/

**Comment by allouch89** [comment]
Every LR channel i have tried so far is not working

**Comment by Shani-08** [comment]
does it work from network with issues or it has sound problems?
If i get time, i think i could look into the latest version to see if i could patch so that you could run it anywhere.

**Comment by allouch89** [comment]
My bad about the LR channels, its required to add session while HD isnt required.
It have sound problems.

**Comment by allouch89** [comment]
Is it possible to change the port of the proxy?

**Comment by emadmahdi** [comment]
@allouch89 @testiddd 

1- the gl proxy version 1.6 does not accepts requests from machines other than localhost (127.0.0.1), so it can not be used for android or any other platforms ... try using version 1.4 ... but you may have audio problems with some channels.

2- gl proxy port cannot be changed unless you have a udp proxifier installed on the same pc that have the gl proxy then in this case you can forward the streams from any platforms and from any ip and from any port to the local machine plus in this case you can use the proxy version 1.6

2- the quality _High is suppose to be sub channel not the main channel ... as for Rotana channels ... to my knowledge gl called them internet links which mean they do not generate the streams they just provide links to the channels ... anyways I will take a look at it ... thanks for the info.

4- sessionID is not required for all gl servers ... some of the servers do accept requests without any sessionID such as these four ip's 19, 20, 42, & 44.

**Comment by emadmahdi** [comment]
@Shani-08 ... I don't think we need to split the gl file ... because currently the non-proxy method does not have lots of options ... it is just available as LOW and for some channels not even all channels.

As for the settings page ... i suggest  removing the enable and disable of the proxy because it is not needed ... check the gl file it do have both methods and both are working without any issues ... plus for people who have gl proxy they can use both at the same time ... and for people who does not have the proxy then they should not play the channels that say ... GLarab : Proxy XXXX

As for the proxy ip in the settings page ... yes it should changed to be a variable that we can use inside the gl file ... same thing applies for the ip of the LR quality.

**Comment by emadmahdi** [comment]
@Shani-08 ... I wonder if we can separate ip's to four types ... I mean something like this:

1- ip for non-proxy method with a username
2- ip for non-proxy method without username
3- ip for proxy method with a username
4- ip for proxy without a username

For example ... we can add the port number 7777 to the ip in the settings list to state that this ip is an ip for a non-proxy method ... and if no port then it means an ip for the proxy method.

Check this link for a full list of ip's ... https://github.com/Shani-08/ShaniXBMCWork/commit/7e07e56b7856df7a385aeee473ee252982363b38

As for the Try All option ... i think it is a good idea if it can be applied to both proxy & non-proxy methods ... since as you know there is ip's that is reserved for premium channels and others reserved for ordinary channels.

**Comment by allouch89** [comment]
I dont know about that _High is a sub channel,
alot of channels doesnt exist as _HD but they do exist as _High

**Comment by Shani-08** [comment]
ok lets take a step back, dont make any changes, and let me think about retry logic. we can setup a section in xml which will just be about ip list, type of channels it supports and requires session or not.
we might need a chat session to solve this easily, i have to find a time for that but will arrange something if we need to sit down and talk about this.
For time being, could you please move the non proxy lines at the top, people who dont have proxy and setup first available channels are complaining that it tries proxy even though its not defined.
we will then solve this one way or another.

**Comment by emadmahdi** [comment]
@Shani-08

No problem i will change the gl file as soon as i finish adding the _High ... give me about an hour ... as for the ip's i think you are moving in the right direction ... it is better to have a table that describes the specs of each ip since gl have a total of 14 ip's but they are serving a total of 46 combinations of urls.

As for the _High I already did a scan and i got the ip's that works for it and they are exactly like the ip's for _HD ... but I am not sure if _HD & _High are suppose to complete each other or they do intersects ... so for now ... I will include both qualities for each channel then we can filter them later.

**Comment by emadmahdi** [comment]
@Shani-08 ... the non-proxy urls are already on top of the gl file ... same for the ip's of the settings page ... i did not change them ... so i just added _High to the end of the gl file.

i think people who have problems using the non-proxy _LOW they need to make sure they are using the ip's that is for the non-proxy method ... these are 42, 44, 26 & 28 ... although the Try All is also ok ... i just tested the non-proxy _LOW with Try All and with the ip 42 and they both works correctly.

**Comment by emadmahdi** [comment]
or may be some users are trying to play channels that are not available for the non-proxy LOW channels ... because currently the gl file have all the channels regardless if it is available or not.

**Comment by emadmahdi** [comment]
I removed the port number 7777 from the settings page for the LOW ip ... it was causing some conflicts because it was written twice at the same time ... one with 7777 and one without 7777 ... it might be the problem that some user were complaining from.

**Comment by allouch89** [comment]
You should not care about required session, just add session for every IP, required or not, it doesnt hurt to add it. Who knows, they might require it later on anyway.

I have now a full list of available channels in _High, _HD, _Med, _Low and _LR.
Let me know if you need it.

**Comment by allouch89** [comment]
X	38.99.146.42
	38.99.146.44

These work with proxy method for _Low channels aswell.

**Comment by allouch89** [comment]
I have noticed that all 'Limited' channels is available as _LR
and all 'Premium' channels is not available as _LR, and not necessary available as _High, _HD, _Med and _Low

**Comment by Shani-08** [comment]
BTW, i invite you guys to read this last couple of pages. You will be amazed as all of your hard work is taken for granted and then complain that its not working!
http://forum.kodi.tv/showthread.php?tid=188114&page=25
Bring popcorns and drinks!

**Comment by allouch89** [comment]
Hehe,
Maybe we should remove the proxy method from shahidmbcnet and create a new plugin just for the glarab proxy.

**Comment by Shani-08** [comment]
I think its about proxy method nor about any source, the guy is just pissed off and throwing toys out of pram. Thanks god i am not charging him money to use the addon, the customer support would've been very busy with one customer, by the look of it :)

**Comment by Shani-08** [comment]
I am making couple of quick changes.
0. there is a new setting for LR ip list 
1. If proxy is disabled then we will not show the proxy sources when you click play.
2. if proxy is enabled then you can play both the proxy and non proxy depending what you click.
3. you disable any ip, it will disable for both the proxy and non proxy, like if you disable HD then it will disable HD links
4. HD and High both are using same ip list
5. i will be using the link defined in the xml for proxy, if variable for ip is used, it will use the user settings, if its Try All, then it will use first ip (as there is no try all) so make sure the first ip should be working with proxy, otherwise use the ip directly in the xml.
i will push these changes in 15 mins, let me know

**Comment by emadmahdi** [comment]
This guy on the forum he is just acting like a child ... he wants the toy but he does not know how to ask for it ... and on top of that he is complaining that the toy is not good for him ... then he start to complain that the toy does not looks legitimate in his mind ... that is funny !!!!! 

Personally I don't care ... I will continue doing what I am doing if someone does not like it then it is his problem not mine.

**Comment by Shani-08** [comment]
update text in my previous comments

**Comment by Shani-08** [comment]
Actually 2M medium getting stuck regardless but previously it was using the session-ip and for high we used to get first ip. let me know if you think otherwise

**Comment by emadmahdi** [comment]
I've seen this stack in other channels but usually it goes by changing the ip for that quality.
The gl channels are really complex ... the use three level of channel categorization ... Ordinary, Premium, Premium Links ... if wrong ip is used it might get stacked.

**Comment by Shani-08** [comment]
It was stucking due to a bug :)
btw, look on another issue i created, need to act on it asap.

**Comment by hawwach** [comment]
@Shani-08 that guy from the forum is a child indeed, instead of being grateful, he's acting like an *****. Best thing to do is ignoring such ignorance.
Thumbs up for your great work.
Thanks.

**Comment by Shani-08** [comment]
don't worry, i was just amazed how his mind works! he win hands down in being the most stupid person i ever come across!

**Comment by emadmahdi** [comment]
@Shani-08 you need to have some Facebook communication with strangers and then you will really be surprised about how stupide some people are.

**Comment by Shani-08** [comment]
you know, i work closely with support and clients who would say anything and people who would just say "This doesn't work". I have a lot of patience so it takes a long time to wind me up but he was very close to that :)
btw, i have release 3.5.0 , now with proxy blocked, it will only show low, with on, you can play both.
the title tell us what type of entry is that so it should only have,
For non proxy
High
HD
Low
Med
LR

For proxy types
Proxy LR
Proxy High
Proxy HD
Proxy Med
Proxy Low

**Comment by emadmahdi** [comment]
@Shani-08 what are the variables for the Proxy & for _LR ... I want to update the gl file.

**Comment by Shani-08** [comment]
$GL-IPLR$

**Comment by Shani-08** [comment]
for proxy, i haven't created yet, so leave it as 127

**Comment by emadmahdi** [comment]
@Shani-08 ... I did not understand your question.

another thing ... since you are limiting the channels ... can you also limit the ip list or may be force it to default values ?

**Comment by emadmahdi** [comment]
or may be since for the non-proxy method we have only LOW works ... I suggest splitting the ip list for the LOW to be two lists ... one for proxy method and the second for non-proxy method ... otherwise we can remove the ip's that works for non-proxy alone and keep only the ip's that works for both methods.

ip's 42 & 44 works for both methods.
ip's 26 & 28 works only for non-proxy method.

**Comment by Shani-08** [comment]
For proxy method I am now using first ip for all qualities now. 
So as long as the first ip is working for proxy, I don't mind what else you add after. 
We can turn off ip selection from user and It will be try all for everyone. 
You just need to make sure the values are updated in the file and assume it's try all selected by the user.

**Comment by emadmahdi** [comment]
@Shani-08 please check your yahoo email ... important.

**Comment by emadmahdi** [comment]
gl file is clean and ready as follows:

146 Proxy High
167 Proxy HD
166 Proxy Med
162 Proxy Low
151 Proxy LR
159 Low

951 Total working url's

**Comment by emadmahdi** [comment]
@Shani-08 ... i think it is time to separate the ip list of the LOW to two lists ... one for proxy low and another for the non-proxy low ... because right now all the 4 ip's of the non-proxy Low are Ordinary channels while all the other ip's which belongs to the Proxy Low are premium ... hence currently only one of them can be available not both ... either low or proxy low ... by separating them then both can be made available at the same time.

It works like this now ... if a user selected an ip of a non-proxy Low then he will not be able to watch any premium channels even by using a proxy low channel which suppose to play the premium ... and if the user selected an ip for Proxy Low so he can get the premium channels to work then all the non-proxy low channels will not work.

**Comment by Shani-08** [comment]
But if user select Try all then It should work right, provided we out first up which works with proxy?
What I am thinking is to use their mobile app calls, I don't even need ip as GL will tell us right up for low which I tested and don't need login details as well. Hence all ips will be for proxy.

**Comment by Shani-08** [comment]
Thanks

**Comment by emadmahdi** [comment]
@Shani-08 ... no even with Try All ... the premium is still not working.

**Comment by emadmahdi** [comment]
@Shani-08 you are welcome.

i forgot to mention that we are now covering a total of 315 unique channels.

**Comment by Shani-08** [comment]
OK -  Good news is that the proxy retry logic wasn't hard and its already coded. click download. This will also show you which ip finally selected for both the proxy and non proxy mode.
It will try all the ip, including the one session sent back.
Now we just need to make sure that whatver is there for proxy and non proxy should work most of the time, if it doesnt then we need to remove it. 
I think @emadmahdi  already done a lot of work and @allouch89  thanks for contributing but please try and see if its possible to keep this clean and remove or update title to text with [x]
so for proxy HD it should be
Proxy HD [x]
or for non proxy
Low [x]

**Comment by allouch89** [comment]
i dont understand what you mean by [x]

**Comment by Shani-08** [comment]
I am saying that if a quality stops working, you can put X to it so that users not this working. we dont need to remove it if it used to work.

**Comment by emadmahdi** [comment]
good trick @Shani-08 ... i did not try the [x] yet ... but your ideas always surprise me !!!!

**Comment by Shani-08** [comment]
http://178.33.241.201:4500/channel.flv?server=8.21.48.20&channel=RotanaAflam_High

**Comment by allouch89** [comment]
8.21.48.19, 8.21.48.20 and  38.99.146.158 aint working anymore.

**Comment by emadmahdi** [comment]
@Shani-08 the problem of the LOW is still there for both proxy & non-proxy ... and same problem even with Try All ... would you please separate the LOW to two lists ... or may be you can limit the ip list when proxy is enabled. the other qualities are working fine ... except the LOW it has conflicts in ip's ... I downloaded the new files and the new version 3.5.1 ... but still same problem.

**Comment by Shani-08** [comment]
Both the proxy and non proxy will try all ips. Click on download and it should work. I'll saparate it anyways.

**Comment by Shani-08** [comment]
You don't have to change url. Both the low and proxy low will have same variable. But the code know which settings to use. The setting will be in setting xml. Will push my changes in 30 mins.

**Comment by Shani-08** [comment]
I have updated the XML and code to use the proxy ip setup in the settings. I was told that with proxy glarab_http_proxy-1.4.0.5571 it was working fine on android systems using windows as local proxy. In the xml, please use $GLProxyIP$ instead of 127.0.0.1 ( i have done that)

**Comment by emadmahdi** [comment]
Thank You @Shani-08

Everything looks ok now ... except gl they are working on some ip's but the Try All is automatically fixing their changes.

Also ... the two lists of ip's for the LOW .... both are having the same label in the settings page ... it is not known which list is for the proxy method and which list is for non-proxy method.

As for the proxy 1.4.0.5571 ... I will install it and test it.

**Comment by Shani-08** [comment]
Yes I was little lazy. You can create a new label in langauge file and enter the id to the settings. The second one is non proxy.

**Comment by emadmahdi** [comment]
I added the new label ... but the language file is not included in files download.

i noticed the LOW is doing lot's of Try All ... so i checked my Kodi log file and noticed that the list for the LOW is always the proxy list ... while the non-proxy list of the LOW is not used at all.

**Comment by allouch89** [comment]
I dont know if you have noticed it but we only have 1 ip left that is working for HD/High channels.
I guess its only a matter of time before we have none.

**Comment by allouch89** [comment]
1 minute later, the last working ip isnt working anymore..

**Comment by Shani-08** [comment]
@emadmahdi good spot, i fixed that bug last night and send you guys response saying i've fixed it yet i forgot to checkin and subsequently lost my change lol :)
now click download and it should get the updated code, the Non proxy should work and pickup correct address. Right now the HD servers are not playing the game but everything else is good. 
I thought we had a guys helping us to find the communcation between the box and server?

**Comment by allouch89** [comment]
We might aswell give up, everything about glarab that gets published here, gets shut down after a few days anway..

**Comment by Shani-08** [comment]
then stop publishing, setup a google group of something and only allow few people in and do the talk there

**Comment by hassuni** [comment]
@emadmahdi  did you received my email?

**Comment by allouch89** [comment]
good idea @Shani-08 but the everything that is published can be accessed by anyone, anyway.
we dont only need a group to discuss stuff, we also need to be able to hide some stuff like the IP's etc.

**Comment by Shani-08** [comment]
its a difference between making it easy and difficult. in anycase posting ip is not helpful.
i can make settings section encrypted but we have to make it invisible to the user and treat the settings is "Try All"

**Comment by allouch89** [comment]
thats acceptable, as long as we can get it to work for more than a few days, then it will be worth it.
after all, its better than nothing. if the user dont like it, he can always try other sources.

**Comment by hassuni** [comment]
you can see here what  Vdubt25 addon are doing,
they ara using Glarab ip as low quality but you can not see it on pastebin, you need to play every channels to see the link, they make it harder
 http://pastebin.com/raw.php?i=D1mC4Fb5

**Comment by Shani-08** [comment]
@hassuni  its not hard but you need to know what to do ;) let me know if you want to see all the links?

**Comment by hassuni** [comment]
@Shani-08 it will be nice if I can see the link, shaum me how I can see it. thanks 
secound , How you can find the playpath of this channel, I used streamtransport i found (nat2sd) and I gussed they have nat2hd.
http://streaminghd.tn/index.php/watanya-2.html

**Comment by testiddd** [comment]
that base64 decode is really hard , even notepad++ can do that out of the box :)

**Comment by emadmahdi** [comment]
It looks like all gl premium or hd channels had stopped working ... i guess it is a session issues or may be ports issues because ip's cannot be changed that fast while settings can be changed easily.

Any one have any idea about the new changes ??!!!!!

Also ... i don't think it is useful to hide links or ip's or decrypt them ... because for gl or any other streaming company they know that ... they need to take an action whenever they see the channels are working using a tool or a software or a method that they did not authorize.

@hassuni ... yes i received your email ... but before we start implementing the new source we first need to make sure that their service is reliable enough plus useful ... i think we are adding more and more sources and spending more time and efforts on developing but then after little while it become useless.

**Comment by emadmahdi** [comment]
i think it will be more efficient & useful if we hide the name of the sources ... i mean we just play the channels without letting the user knows it belongs to what source.

**Comment by Shani-08** [comment]
@emadmahdi session comes after when you are able to connect. I tried to look into those servers and they reject the connection from telnet on port 4500 so i think they are down and moved out.

**Comment by testiddd** [comment]
8.21.48.151 / 187 / 188 / 253 Port 7777 open , connection oki , but then either failed to read answer or 404

**Comment by testiddd** [comment]
on that range 38.99.146.1 - 38.99.146.254 ... Port 7777 open on 71/103/151 , but same situation ... think we need a log to get by.

**Comment by emadmahdi** [comment]
@testiddd ... the ip's you are listing is for gl but not for arabic language .... it might be any other language ... gl serve 6 languages.

**Comment by emadmahdi** [comment]
@Shani-08 ... in this case it might be a port number changes.

**Comment by Shani-08** [comment]
thing is we all trying to guess, we need glbox to be certain, i thought there were someone with the box? that will close the whole lot of pain

**Comment by emadmahdi** [comment]
we also need to know how the box is updating the list of channels or updating the settings.

**Comment by Shani-08** [comment]
Click download and there is a surprise for you :)

**Comment by allouch89** [comment]
Who is the man? You are the man! GJ

**Comment by norhan76** [comment]
Thank u so much for your efforts guys. I tried it on windows and working perfect but doesn't on my android box. I'm not sure why.

**Comment by Shani-08** [comment]
On android, you need to have a proxy that is available on network and working and make sure the settings are pointing to that machine. Make sure the version of the proxy is that particular one which allow network usage. Look up somewhere and you'll find the version.

**Comment by norhan76** [comment]
Sorry for bothering u, but how can i do that. Is there any steps i can follow.

**Comment by hassuni** [comment]
just install Proxy http_proxy-1.4.0.5571.exe
https://www.glarab.com:447/zixi/download/proxy/glarab_http_proxy-1.4.0.5571.exe

go to pc and open cmd.exe and type ipconfig
look for ip4v4v address. should look like this 192.168.1.1
go to shahid addon setting AND type  your localhost ip adress ( IPv4)

**Comment by norhan76** [comment]
Thank u so much guys.

On Sun, Feb 8, 2015 at 1:09 PM, hassuni <notifications@github.com> wrote:

> just install Proxy http_proxy-1.4.0.5571.exe
>
> https://www.glarab.com:447/zixi/download/proxy/glarab_http_proxy-1.4.0.5571.exe
>
> go to pc and open cmd.exe and type ipconfig
> look for ip4v4v address. should look like this 192.168.1.1
> go to shahid addon setting AND type your localhost ip adress ( IPv4)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Shani-08/ShaniXBMCWork/issues/161#issuecomment-73423108>
> .
>



-- 
Norhan Hammad
Web/Graphic Designer
Phone: (302)743-7697

**Comment by mazen313** [comment]
Please i have an atv2 with kodi installed on it. After the last update of the plugin, most of the sources are not appearing. Any solution for the problem ?

**Comment by Shani-08** [comment]
closing this issue as well as this served the purpose.
